### PR TITLE
chore: misc clean up in forest_interpreter

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- [x] src/vm.rs: Remove the unused Heights parameter.
- [x] src/vm.rs: Modify circ_supply_calc to take a database argument instead of a StateTree argument.
- [x] src/default_runtime.rs: Move resolve_to_key_addr to src/lib.rs.
- [x] src/fvm/externs.rs: Move to src/fvm.rs.

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/forest/issues/2111

**Other information and links**

### CODE REVIEW

File: vm/interpreter/src/vm.rs [hunk 1]
```diff
@@ -99,16 +97,13 @@ where
...
         engine: Engine,
-        heights: Heights,
         chain_finality: i64,
     ) -> Result<Self, anyhow::Error>
     where
         R: Rand + Clone + 'static,
-        C: FnOnce(ChainEpoch, &StateTree<&DB>) -> Result<TokenAmount, anyhow::Error>,
+        C: FnOnce(ChainEpoch, &DB) -> Result<TokenAmount, anyhow::Error>,
     {
-        let state = StateTree::new_from_root(&store_arc, &root)?;
```
Comment (hanabi1224) [inline vm/interpreter/src/vm.rs:109]: @lemmih Looks like it's possible to totally remove this callback and pass `circ_supply` as fn arg

File: blockchain/state_manager/src/lib.rs [hunk 2]
```diff
@@ -359,13 +358,15 @@ where
                 &rand_clone,
                 base_fee.clone(),
                 network_version,
-                |height, state| self.genesis_info.get_circulating_supply(height, state),
+                |height, db| {
+                    let state = StateTree::new_from_root(db, &state_root)?;
+                    self.genesis_info.get_circulating_supply(height, &state)
```
Comment (lemmih) [inline blockchain/state_manager/src/lib.rs:363]: Could you move `new_from_root` into `get_circulating_supply`?

File: vm/interpreter/src/vm.rs (hunk 1 above)
Comment (lemmih) [reply vm/interpreter/src/vm.rs:109]: Go for it.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: vm/interpreter/src/vm.rs (hunk 1 above)
Comment (hanabi1224) [reply vm/interpreter/src/vm.rs:109]: Done.

File: blockchain/state_manager/src/lib.rs (hunk 2 above)
Comment (hanabi1224) [reply blockchain/state_manager/src/lib.rs]: Done.

File: utils/metrics/Cargo.toml [hunk 3]
```diff
@@ -10,4 +10,4 @@ fs_extra            = "1.2"
 log.workspace       = true
 prometheus          = { workspace = true, features = ["process"] }
 thiserror.workspace = true
-tide.workspacce     = true
```
Comment (hanabi1224) [inline utils/metrics/Cargo.toml:13]: happen to find a typo

**Review by lemmih — APPROVED** [review]

**Review by connormullett — APPROVED** [review]

**Pull request merged**